icu4c/source/i18n/unicode/tblcoll.h

```diff
@@ -245,6 +245,25 @@ class U_I18N_API RuleBasedCollator : public Collator {
     virtual CollationElementIterator* createCollationElementIterator(
                                            const UnicodeString& source) const;
 
+#ifndef U_HIDE_INTERNAL_API
+     /**
+     * Creates a collation element iterator for the source string. The caller of
+     * this method is responsible for the memory management of the returned pointer.
+     * 
+     * This method is like the above createCollationElementIterator, but instead uses
+     * a read-only alias of the input source string.
+     * The input source string must outlive the returned object's lifetime.
+     * 
+     * @param source the string over which the CollationElementIterator will
+     *        iterate.
+     * @return the collation element iterator of the source string using this as
+     *         the based Collator.
+     * @internal
+     */
+    virtual CollationElementIterator* createCollationElementIteratorReadOnlyAlias(
```

Comment:
Does it have to be virtual? If it does, you can't HIDE it because you would create an incompatible vtable.

I think a good alternative is to not add anything here. Change call sites to pass in an empty string and then call the new text setter. We can look at optimizing a CEI with no string / an empty string.